From 4328ad7456b3c0e9a1863924322244fa28c459ec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jo=C3=A3o=20Henrique?= <johnnyonflame@hotmail.com>
Date: Sun, 14 Jan 2024 04:39:56 -0300
Subject: [PATCH 4/5] Fix DRM_FORMAT_A/XRGB8888 swizzle for libgo2.

Previously this was defined as RK_FORMAT_BGRA_8888, which is the wrong channel layout for these formats.
This commit fixes wrong color output on devices such as the RG Arc.
---
 deps/libgo2/src/display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/libgo2/src/display.c b/deps/libgo2/src/display.c
index 16b6a9f8e3..1b8246d585 100644
--- a/deps/libgo2/src/display.c
+++ b/deps/libgo2/src/display.c
@@ -550,7 +550,7 @@ static uint32_t go2_rkformat_get(uint32_t drm_fourcc)
 
         case DRM_FORMAT_ARGB8888:
         case DRM_FORMAT_XRGB8888:
-            return RK_FORMAT_BGRA_8888;
+            return RK_FORMAT_RGBX_8888;
 
         case DRM_FORMAT_RGB565:
             return RK_FORMAT_RGB_565;
-- 
2.43.0

